### ISSUE TITLE
Use parse5's tree adapter directly instead of the current indirection

### ISSUE THREAD

**Issue by domenic** [tmpvar/jsdom#1851]
My understanding of what we currently do, in the htmltodom.js file, is:

- We use the htmlparser2 tree adapter built into parse5 to get a particular representation of the nodes. This is done because historically we used to use htmlparser2, so this was an easy transition path.
- We then use this function named `setChild` to convert the htmlparser2 representation into actual jsdom classes

This is roundabout and unnecessary. The `setChild` function is full of strange hacks around templates, doctypes, and namespaced attributes that would probably be unnecessary if we just implemented a proper jsdom tree adapter for parse5.

We already have half of a tree adapter, in `documentAdapter.js`, which we use for serialization. We should finish it out and try to use it.

/cc @Sebmaster in case he remembers any different details.

**Comment by Sebmaster** [comment]
Doing our own _mostly_ works, except for scripts. We'll create an empty script tag, append that, then get the scriptText from parse5 and HTML does not re-execute on changed nodeText of scripts. So we'd have to hack around that and trigger another execution in that case.

**Comment by domenic** [comment]
Right, I see, and at that point it's tempting to skip to doing things the "right" way with the streaming document.write()-aware version... but that's hard, of course.
